lib/scrabble/scrabble.rb

```diff
@@ -0,0 +1,84 @@
+module Scrabble
+  class Scrabble
+
+    SCORING= {a: 1, b: 3, c: 3, d: 2, e: 1, f: 4,
+      g: 2, h: 4, i: 1, j: 8, k: 5,
+      l: 1, m: 1, n: 1, o: 1, p: 3,
+      q: 10, r: 1, s: 1, t: 1, u: 1,
+      v: 4, w: 4, x: 8, y: 4, z: 10}
+
+    def self.score(word)
+      #accounting for blank user input
+      return 0 if word == ""
+      return 0 if word == nil
+      #accounting for a single digit non alphabetical character
+      if word.to_i != 0 || word == "0"
+        puts "Invalid input."
+        return 0
+      end
+      if word.length > 7
+        puts "Word length is limited to 7 letters."
+        return 0
+      end
+      #splitting the word into an array of lowercase letters
+      letters = word.downcase.split(//)
+      #accounting for a multicharacter response that includes a non alphabetical character
+      letters.each  do |x|
+        if x.to_i != 0 || x == "0"
+          puts "Your word contains an invalid character."
+          return 0
+        end
+      end
+
+      score = 0
+      letters.each do |letter|
+        score += SCORING[letter.to_sym]
+      end
+      return score
+    end
+
+     #returns the word in the array with the highest score.
+    def self.highest_score_from(array_of_words)
+      word_scores = array_of_words.group_by do |word|
+        score(word)
+      end
+
+      high_score = word_scores.keys.max
+
+      if word_scores[high_score].length > 1
+        shortest, longest = word_scores[high_score].minmax_by do |word|
+          word.length
+        end
+
+        if longest.length == 7
```

Comment:
I see you're using the number "7" a couple times here. Totally fine, of course, but if someone wasn't clear about 7 being the longest length of a scrabble word, it could confuse them.. and if you needed to change it to 8 letters later on, you have to remember to change it in both places. A way to get around this is to declare a constant with a clear name that explains what it is, then use that constant instead of a "magic number"

```ruby
MAX_SCRABBLE_WORD_LENGTH = 7
```

up at the top, and then here you'd do this:

```ruby
if longest.length == MAX_SCRABBLE_WORD_LENGTH
```